velero-plugins/serviceaccount/restore.go

```diff
@@ -0,0 +1,67 @@
+package serviceaccount
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/heptio/velero/pkg/plugin/velero"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// RestorePlugin is a restore item action plugin for Velero
+type RestorePlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to everything
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"serviceaccounts"},
+	}, nil
+}
+
+// Execute fixes the route path on restore to use the target cluster's domain name
+func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("[serviceaccount-restore] Entering ServiceAccount restore plugin")
+
+	serviceAccount := corev1.ServiceAccount{}
+	itemMarshal, _ := json.Marshal(input.Item)
+	json.Unmarshal(itemMarshal, &serviceAccount)
+
+	p.Log.Info("[serviceaccount-restore] Checking for pull secrets to remove")
+	check := serviceAccount.Name + "-dockercfg-"
+	for i := len(serviceAccount.Secrets) - 1; i >= 0; i-- {
```

Comment:
Wonder if using `range` here would be cleaner for defining the loop. https://gobyexample.com/range

Reply:
I followed the example of velero upstream -- but I think the explicit countdown is safer here since we're modifying the underlying slice in the loop.

Reply:
+1